llms/mlx_lm/utils.py

```diff
@@ -113,11 +138,21 @@ def sample(logits: mx.array) -> Tuple[mx.array, float]:
         prob = softmax_logits[0, token]
         return token, prob
 
+    if repetition_penalty < 0 or not isinstance(repetition_penalty, float):
+        raise ValueError(
+            f"repetition_penalty must be a non-negative float, got {repetition_penalty}"
+        )
+
     y = prompt
     cache = None
+
     while True:
-        logits, cache = model(y[None], cache=cache)
-        logits = logits[:, -1, :]
+        logits, cache = model(y[:, None], cache=cache)
+        logits = logits.squeeze(1)
```

Comment:
Issue: Revert `logits = logits.squeeze(1)` back to `logits = logits[:, -1, :]` because we only return the last token from the generate step instead of the whole sequence.